packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.spec.ts

```diff
@@ -648,6 +648,174 @@ describe("prismaSchemaParser", () => {
           expect(result).toEqual(expectedEntitiesWithFields);
         });
 
+        it("should add the @id attribute to a field if its name is id and there is id and @@id attribute on the model", async () => {
```

Comment:
I'm not sure we should change the primary key as we would cause data loss (it would loose the relationships during migration). 
Moreover, this will unlikely going to allow successfully db migrations for the reason above.

To validate this, try the following:

1. run migrate on 
```
datasource db {
  provider = "postgresql"
  url      = env(DB_URL)
}

generator client {
  provider = "prisma-client-js"
}

model Doctor {
  id        Int       @default(autoincrement())
  firstName String
  lastName  String

  @@id([firstName, lastName])
}
```
 

2. add a new model run migrate on 
```
datasource db {
  provider = "postgresql"
  url      = env(DB_URL)
}

generator client {
  provider = "prisma-client-js"
}

model Doctor {
  id        Int       @default(autoincrement())
  firstName String
  lastName  String
  Patient   Patient[]

  @@id([firstName, lastName])
}

model Patient {
  id              Int    @id @default(autoincrement())
  firstName       String
  lastName        String
  doctorFirstName String
  doctorLastName  String
  doctor          Doctor @relation(fields: [doctorFirstName, doctorLastName], references: [firstName, lastName])
}
```

3. add a new doctor and a new patient in the db
4. update the schema with the expected amplication generated schema and run db migrate:
```datasource db {
  provider = "postgresql"
  url      = "postgres://admin:admin@localhost:5432/tmp"
}

generator client {
  provider = "prisma-client-js"
}

model Doctor {
  id        Int       @id @default(autoincrement())
  firstName String
  lastName  String
  Patient   Patient[]

  @@unique([firstName, lastName])
}

model Patient {
  id              Int    @id @default(autoincrement())
  firstName       String
  lastName        String
  doctorFirstName String
  doctorLastName  String
  doctor          Doctor @relation(fields: [doctorFirstName, doctorLastName], references: [firstName, lastName])
}
```

Reply:
So we may have a bigger problem because we took the same approach when there is an @@id attribute on the model, and there is no Id field at all. In this case we decided to add an id field to the model in order to match the dsg requirements